app/services/gps_report.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+class InvalidSupplierError < StandardError; end
+
+class GPSReport
+  include ActionView::Helpers::TextHelper
+
+  def initialize(date_range, supplier_name)
+    @supplier = Supplier.find_by(key: supplier_name)
+    raise InvalidSupplierError, "Invalid supplier: `#{supplier_name}`. Valid values are: `#{Supplier.all.map(&:key).join('`, `')}`." if @supplier.nil?
```

Comment:
```suggestion
    raise InvalidSupplierError, "Invalid supplier: `#{supplier_name}`. Valid values are: `#{Supplier.pluck(:key).join('`, `')}`." if @supplier.nil?
```

I think this avoids fetching all the fields from the database when we only need the `:key`.